src/main/java/com/emeq/counter/service/FoodService.java

```diff
@@ -0,0 +1,70 @@
+package com.emeq.counter.service;
+
+import java.net.URI;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+import com.emeq.counter.dto.FoodDTO;
+import com.emeq.counter.exception.RecordNotFoundException;
+import com.emeq.counter.model.Food;
+import com.emeq.counter.repository.FoodRepository;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+
+@Service
+public class FoodService {
+
+    @Autowired
+    private FoodRepository foodRepository;
+
+    private static final String EXCEPTION_STRING = "Invalid Food Id : ";
+
+    public ResponseEntity<List<FoodDTO>> index() {
+        Iterable<Food> foods = foodRepository.findAll();
+
+        List<FoodDTO> result = StreamSupport.stream(foods.spliterator(), false).map(Food::convert2DTO)
+                .collect(Collectors.toList());
+
+        return ResponseEntity.ok(result);
+    }
+
+    public ResponseEntity<FoodDTO> show(final Long id) {
+        Food food = foodRepository.findById(id).orElseThrow(() -> new RecordNotFoundException(EXCEPTION_STRING + id));
+
+        return ResponseEntity.ok(Food.convert2DTO(food));
+    }
+
+    public ResponseEntity<FoodDTO> create(final FoodDTO foodDTO) {
+        final Food food = new Food();
+        food.setName(foodDTO.getName());
+        FoodDTO result = Food.convert2DTO(foodRepository.save(food));
+
+        URI location = URI.create("http://localhost:3000/foods/" + result.getId());
```

Comment:
I think we should get url body from some kind of config that we can override per environment, deployment.
Otherwise this hardcoded url is going to give us trouble when trying to deploy to different environments e.g. heroku.

Reply:
I added three environment to application. dev, test and prod. Dev is default environment. When run tests, environment will changes.